### PR TITLE
Add Stock Price Predictor

### LINKED ISSUE #328: 💡[FEATURE]: Stock Price Prediction
**Is your feature request related to a problem? Please describe.**

Yes, accurately predicting stock prices can help investors make informed decisions, potentially increasing returns and minimizing risks in the stock market. However, it is challenging due to the volatile and unpredictable nature of financial markets.

**Describe the solution you'd like**

I would like to develop a machine learning model using historical stock price data and relevant financial indicators. This model should predict future stock prices with high accuracy, providing actionable insights for investors.

**Describe alternatives you've considered**

Alternatives considered include using traditional statistical methods, technical analysis, or sentiment analysis from news and social media. However, these methods may lack the predictive power and adaptability of a machine learning-based approach.

**Additional context**
Screenshot of Tata Motors Price Prediction:

Screenshot 2024-06-08 at 1 37 16 PM
Screenshot 2024-06-08 at 1 38 08 PM

### PR DESCRIPTION
**Pull request by vishuhere**
## 🛠️ Related Issue

- Closes #328 

## 📝 Description

Stock Price Prediction Using RNN

Overview Of Project:

Stock Price Prediction using machine learning helps to discover the future value of company stock on stock market. The benefit of predicting the value of stock is to gain more profit from the market. For the closing price prediction of Tata Motors, the model is built using Long Short Term Memory Network (LSTM). The model is trained using the past records of the Company from Yahoo Finance.

Libraries/Frameworks used:

pandas_datareader: It is used to extract the dataset of stock prices.

numpy: It is used to handle the arrays.

pandas: It is used to handle data frames.

MinMaxScaler: Is used to scale the data.

Sequential: To build the ML model

Dense: It is a layer for the model, in which each neuron receives input from all the neurons of the previous layer.

LSTM: It is a layer for the model, in which each neuron receiver input from all the past previous layers.

Dropout: It is a layer for the model, it helps to randomly select the neurons and helps to reduce the overfitting of the model.

matplotlib.pylot: Is used for plotting the graphs for values.

## 🔍 Type of PR
- [ ] ✨ Feature enhancement

## 📸 Screenshots / 🎥 Videos (if applicable)

Screenshot 2024-06-08 185401
Screenshot 2024-06-08 185611

## ✅ Checklist

- [  ✅ ] I have performed a self-review of my code.
- [  ✅ ] I have read and followed the Contribution Guidelines.
- [  ✅ ] I have tested the changes thoroughly before submitting this pull request.
- [  ✅ ] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [  ✅ ] I have commented my code, particularly in hard-to-understand areas.

## ℹ️ Additional Context

Dataset link: https://finance.yahoo.com/quote/TTM/history?p=TTM
This dataset contains all the previous record of stock price of Tata Motors Company which was mentioned in my photo.

### CODE REVIEW

**Comment by vishuhere** [comment]
Hey @Akshat111111 that was a fun project. I am looking for some good connections on LinkedIn and I really appreciate your work on such organisation, **I would like to connect over on LinkedIn**. Also If you liked my effort on this project kindly close this issue with updating the label 2 as **label 3 for GSsoC,** It will be really helpful for me.

**Comment by Akshat111111** [comment]
> Hey @Akshat111111 that was a fun project. I am looking for some good connections on LinkedIn and I really appreciate your work on such organisation, **I would like to connect with you over on LinkedIn**. Also If you liked my effort on this project kindly close this issue with updating the label 2 as **label 3 for GSsoC,** It will be really helpful for me.

yeah, sure

**Review by Akshat111111 — COMMENTED** [review]
Rename your folder as tata motors analysis and put it inside NSE main folder

**Comment by vishuhere** [comment]
> > Hey @Akshat111111 that was a fun project. I am looking for some good connections on LinkedIn and I really appreciate your work on such organisation, **I would like to connect with you over on LinkedIn**. Also If you liked my effort on this project kindly close this issue with updating the label 2 as **label 3 for GSsoC,** It will be really helpful for me.
> 
> yeah, sure

Hey @Akshat111111 I had made the changes as you have asked. Also kindly provide me your LinkedIn URL

**Comment by Akshat111111** [comment]
> > > Hey @Akshat111111 that was a fun project. I am looking for some good connections on LinkedIn and I really appreciate your work on such organisation, **I would like to connect with you over on LinkedIn**. Also If you liked my effort on this project kindly close this issue with updating the label 2 as **label 3 for GSsoC,** It will be really helpful for me.
> > 
> > 
> > yeah, sure
> 
> Hey @Akshat111111 I had made the changes as you have asked. Also kindly provide me your LinkedIn URL

its there in my github profile

**Review by Akshat111111 — COMMENTED** [review]
I have told to put your folder inside nifty, so do it
